app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -93,21 +93,15 @@ def show
     param :id, :identifier, :desc => N_("content view numeric identifier"), :required => true
     param :name, String, :desc => N_("module name to restrict modules for"), :required => false
     def available_puppet_modules
-      current_ids = @view.content_view_puppet_modules.map(&:uuid).reject { |p| p.nil? }
+      current_uuids = @view.content_view_puppet_modules.map(&:uuid).reject { |p| p.nil? }
+      repositories = @view.organization.library.puppet_repositories
 
-      repo_ids = @view.organization.library.puppet_repositories.pluck(:pulp_id)
-      search_filters = [{ :terms => { :repoids => repo_ids }}]
+      query = PuppetModule.in_repositories(repositories)
+      query = query.where(:name => params[:name]) if params[:name]
+      query = query.where("#{PuppetModule.table_name}.uuid NOT in (?)", current_uuids.join(',')) if current_uuids.present?
```

Comment:
no need to do the current_uuids.join here,  AR knows how to handle an array of ids